curl-helper.c

```diff
@@ -4593,6 +4597,52 @@ value caml_curl_pause(value conn, value opts)
   CAMLreturn(Val_unit);
 }
 
+#if HAVE_DECL_CURL_EASY_NEXTHEADER
+value caml_curl_get_headers_rev(value conn, value opts, value request)
+{
+  CAMLparam3(conn, opts, request);
+  CAMLlocal2(headers, tuple);
+  Connection *connection = Connection_val(conn);
+
+  int origin = 0;
+  while (Val_emptylist != opts)
+    {
+      switch (Int_val(Field(opts, 0)))
+        {
+        case 0: origin |= CURLH_HEADER; break;
+        case 1: origin |= CURLH_TRAILER; break;
+        case 2: origin |= CURLH_CONNECT; break;
+        case 3: origin |= CURLH_1XX; break;
+        case 4: origin |= CURLH_PSEUDO; break;
+        default: caml_failwith("wrong headerOrigin");
+        }
+      opts = Field(opts, 1);
+    }
+
+  headers = Val_emptylist;
+  struct curl_header *header = NULL;
+  while (NULL != (header = curl_easy_nextheader(connection->handle, origin, Int_val(request), header))) {
+    tuple = caml_alloc_tuple(2);
+    Store_field(tuple, 0, caml_copy_string(header->name));
+    Store_field(tuple, 1, caml_copy_string(header->value));
+    headers = Val_cons(headers, tuple);
+  }
+
+  CAMLreturn(headers);
+}
+#else
+value caml_curl_get_headers_rev(value v_ignored)
```

Comment:
```suggestion
value caml_curl_get_headers_rev(value v_ignored)
```

Not sure having a different prototype than what's told to the OCaml side is correct on all platforms. The compiler could mess up the calling convention.